### PR TITLE
Some operational semantics.

### PR DESCRIPTION
**Pull request by chriseth**
Solves part of #3860

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
@chriseth Ah, I was already preparing a PR for #3860 so I guess I'll stop on that.

File: docs/types.rst [hunk 1]
```diff
@@ -50,13 +50,51 @@ Operators:
...
 
-Division always truncates (it is just compiled to the ``DIV`` opcode of the EVM), but it does not truncate if both
-operators are :ref:`literals<rational_literals>` (or literal expressions).
+
+Comparisons
+~~~~~~~~~~~
+
+The value of a comparison is the one obtained by comparing the integer value.
+
+Bit operations
```
Comment (ekpyron) [inline docs/types.rst]: Maybe we should mention here again that shifts are arithmetic shifts, are considered to be arithmetic operators, not bit operators, and explained below in detail? I would usually consider shifts to be bit operations and a reader might miss the fact that signed right shifts are arithmetic, if it's not mentioned here...

File: docs/types.rst [hunk 2]
```diff
@@ -50,13 +50,51 @@ Operators:
...
+
+Exponentiation is only available for unsigned types. Please take care that the types
+you are using are large enough to hold the result and prepare for potential wrapping behaviour.
+
+Note that ``0**0`` is defined by the EVM as ``1``.
+
+Shifts
+~~~~~~
+
 The result of a shift operation is the type of the left operand. The
```
Comment (ekpyron) [inline docs/types.rst]: Maybe this needs some rewording as well, the result is not a type - ``The type of the result of a shift operation is...`` or ``The result of a shift operation has the type of ...``

File: docs/types.rst (hunk 1 above)
Comment (chriseth) [reply docs/types.rst]: I'll just move the shift operations directly after "bit operations".

**Comment by chriseth** [comment]
Re-ordered a little and split out shift operations. Clarified "type of result" for shifts.

**Review by erak — APPROVED** [review]

**Review by axic — APPROVED** [review]

**Pull request merged**